libs/Wallet/include/Status/Wallet/SavedAddressesController.h

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+#include <QtQmlIntegration>
+
+#include <Helpers/QObjectVectorModel.h>
+
+#include "SavedAddress.h"
+
+namespace Status::Wallet
+{
+
+class SavedAddressesController : public QObject
+{
+    Q_OBJECT
+    QML_ELEMENT
+    QML_UNCREATABLE("C++ only")
+
+    Q_PROPERTY(QAbstractListModel* savedAddresses READ savedAddresses CONSTANT)
+
+public:
+    enum Error
+    {
+        NoneError,
+        SaveAddressError,
+        RetrieveSavedAddressesError
+    };
+    Q_ENUM(Error)
+
+    SavedAddressesController(QObject* parent = nullptr);
+
+    QAbstractListModel* savedAddresses() const;
+
+    Q_INVOKABLE void saveAddress(const QString& address, const QString& name);
+    Q_INVOKABLE void refresh();
+
+signals:
+    void error(Status::Wallet::SavedAddressesController::Error error);
+
+private:
+    using SavedAddressesModel = Helpers::QObjectVectorModel<SavedAddress>;
+    std::shared_ptr<SavedAddressesModel> m_savedAddresses;
```

Comment:
why do we need shared_ptr here? Using `SavedAddressesModel` seems suitable here.

Reply:
I'd be satisfied with unique_ptr but helper function we use provide shared_ptr ( `Helpers::makeSharedQObject `) 
If we want to change it -  we should do it in all places to keep code unified.

Reply:
But what about not using smart pointers at all and keep it as a member directly, just `SavedAddressesModel` type?

Reply:
If I keep it as member directly i should use casting to remove const qualifier in getter function. I'll replace shared_ptr with unique_ptr, is it ok?

Reply:
ok, sure